test/e2e_node/image_pull_test.go

```diff
@@ -0,0 +1,258 @@
+				if pod.Status.Phase == v1.PodRunning || pod.Status.Phase == v1.PodFailed {
+					return true, nil
+				}
+				return false, nil
+			})
+			framework.ExpectNoError(err)
+			err = e2epod.WaitForPodCondition(ctx, f.ClientSet, f.Namespace.Name, pod2.Name, "Failed", 30*time.Second, func(pod *v1.Pod) (bool, error) {
+				if pod.Status.Phase == v1.PodRunning || pod.Status.Phase == v1.PodFailed {
+					return true, nil
+				}
+				return false, nil
+			})
+			framework.ExpectNoError(err)
+
+			events, err := f.ClientSet.CoreV1().Events(f.Namespace.Name).List(ctx, metav1.ListOptions{})
+			framework.ExpectNoError(err)
+			var httpdPulled, httpdNewPulled pulledStruct
+			for _, event := range events.Items {
+				var err error
+				if event.Reason == kubeletevents.PulledImage {
+					if event.InvolvedObject.Name == testpod.Name {
+						httpdPulled, err = getDurationsFromPulledEventMsg(event.Message)
+						framework.ExpectNoError(err)
+					} else if event.InvolvedObject.Name == testpod2.Name {
+						httpdNewPulled, err = getDurationsFromPulledEventMsg(event.Message)
+						framework.ExpectNoError(err)
+					}
+				}
+			}
+
+			// as this is serialize image pulling, the waiting duration should be almost double the duration with the pulled duration.
+			// use 1.5 as a common ratio to avoid some overlap during pod creation
+			if float32(httpdNewPulled.pulledIncludeWaitingDuration/time.Millisecond)/float32(httpdNewPulled.pulledDuration/time.Millisecond) < 1.5 &&
+				float32(httpdPulled.pulledIncludeWaitingDuration/time.Millisecond)/float32(httpdPulled.pulledDuration/time.Millisecond) < 1.5 {
+				framework.Failf("At least, one of the pull duration including waiting %v/%v should be similar with the pulled duration %v/%v",
+					httpdNewPulled.pulledIncludeWaitingDuration, httpdPulled.pulledIncludeWaitingDuration, httpdNewPulled.pulledDuration, httpdPulled.pulledDuration)
+			}
+
+		})
+
+	})
+})
+
+func prepareAndCleanup(ctx context.Context, f *framework.Framework) (testpod *v1.Pod, testpod2 *v1.Pod) {
+	httpdImage := imageutils.GetE2EImage(imageutils.Httpd)
+	httpdNewImage := imageutils.GetE2EImage(imageutils.HttpdNew)
+	testpod = &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "httpd",
+			Namespace: f.Namespace.Name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{{
+				Name:            "httpd",
+				Image:           httpdImage,
+				ImagePullPolicy: v1.PullAlways,
+			}},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	testpod2 = &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "httpd2",
+			Namespace: f.Namespace.Name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{{
+				Name:            "httpd-new",
+				Image:           httpdNewImage,
+				ImagePullPolicy: v1.PullAlways,
+			}},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+
+	ginkgo.By("cleanup images")
+	_ = RemoveImage(ctx, testpod.Spec.Containers[0].Image)
+	_ = RemoveImage(ctx, testpod2.Spec.Containers[0].Image)
+	return
```

Comment:
nit: naked returns are less readable than explicit ones in my opinion. I'd suggest not to use them in functions longer than a couple of lines or, better never use them.